From 63f47da0a68324eed446dcc53717809a87047881 Mon Sep 17 00:00:00 2001
From: kanlihux <kanlix.hu@intel.com>
Date: Wed, 18 Dec 2019 15:02:37 +0800
Subject: [PATCH 38/42] enable GLX_USE_TLS on android

On the Android deqp test when we run
dEQP-EGL.functional.sharing.gles2.multithread.random.textures.gen_delete.15
we will find two thread using one context. It will case gpu hang.
---
 Android.common.mk      | 2 ++
 src/mapi/Android.mk    | 1 +
 src/mapi/glapi/glapi.h | 4 ++--
 3 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/Android.common.mk b/Android.common.mk
index d92294d90339..f20faa828450 100644
--- a/Android.common.mk
+++ b/Android.common.mk
@@ -69,6 +69,7 @@ LOCAL_CFLAGS += \
 	-DHAVE___BUILTIN_CLZLL \
 	-DHAVE___BUILTIN_UNREACHABLE \
 	-DHAVE_PTHREAD=1 \
+	-DGLX_USE_TLS \
 	-DHAVE_DLADDR \
 	-DHAVE_DL_ITERATE_PHDR \
 	-DHAVE_LINUX_FUTEX_H \
@@ -87,6 +88,7 @@ LOCAL_CFLAGS += \
 	-Wno-enum-conversion
 
 LOCAL_CPPFLAGS += \
+	-DGLX_USE_TLS \
 	-D__STDC_CONSTANT_MACROS \
 	-D__STDC_FORMAT_MACROS \
 	-D__STDC_LIMIT_MACROS \
diff --git a/src/mapi/Android.mk b/src/mapi/Android.mk
index 06a764700994..2a9d8d85d6b4 100644
--- a/src/mapi/Android.mk
+++ b/src/mapi/Android.mk
@@ -44,6 +44,7 @@ LOCAL_SRC_FILES := \
 	u_execmem.c
 
 LOCAL_CFLAGS := \
+	-DGLX_USE_TLS \
 	-DMAPI_MODE_GLAPI \
 	-DMAPI_ABI_HEADER=\"$(abi_header)\"
 
diff --git a/src/mapi/glapi/glapi.h b/src/mapi/glapi/glapi.h
index d5d4e0a03a66..2ce77456ffd8 100644
--- a/src/mapi/glapi/glapi.h
+++ b/src/mapi/glapi/glapi.h
@@ -97,8 +97,8 @@ _GLAPI_EXPORT extern __thread void * _glapi_tls_Context
 _GLAPI_EXPORT extern const struct _glapi_table *_glapi_Dispatch;
 _GLAPI_EXPORT extern const void *_glapi_Context;
 
-# define GET_DISPATCH() _glapi_tls_Dispatch
-# define GET_CURRENT_CONTEXT(C)  struct gl_context *C = (struct gl_context *) _glapi_tls_Context
+# define GET_DISPATCH() _glapi_get_dispatch()
+# define GET_CURRENT_CONTEXT(C)  struct gl_context *C = (struct gl_context *) _glapi_get_context()
 
 #else
 
-- 
2.17.1

